### PR TITLE
SQL: Error when CASE is a part of expression #3913

### PR DESCRIPTION
**Pull request by alexey-kravtsov**

### CODE REVIEW

File: src/test/java/com/axibase/tsd/api/method/sql/operator/SqlCaseOperatorTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import static com.axibase.tsd.api.util.TestUtil.TestNames.entity;
+import static com.axibase.tsd.api.util.TestUtil.TestNames.metric;
+
+public class SqlCaseOperatorTest extends SqlTest {
+    private static final String TEST_ENTITY_NAME = entity();
+    private static final String TEST_METRIC_NAME = metric();
+
+    @BeforeClass
+    public void prepareData() throws Exception {
+        Series series = new Series();
```
Comment (vtols) [inline src/test/java/com/axibase/tsd/api/method/sql/operator/SqlCaseOperatorTest.java]: Possibly you can set metric and entity in place `new Sereis(TEST_ENTITY_NAME, TEST_METRIC_NAME)`

File: src/test/java/com/axibase/tsd/api/method/sql/operator/SqlCaseOperatorTest.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        series.setEntity(TEST_ENTITY_NAME);
+        series.setMetric(TEST_METRIC_NAME);
+        series.setData(Collections.singletonList(Mocks.SAMPLE));
```
Comment (vtols) [inline src/test/java/com/axibase/tsd/api/method/sql/operator/SqlCaseOperatorTest.java]: You also can use `series.addData`

File: src/test/java/com/axibase/tsd/api/method/sql/operator/SqlCaseOperatorTest.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+        assertSqlQueryRows("Incorrect query result with CASE operator in expression", expectedRows, sqlQuery);
+    }
+
+    /**
+     *  #3913
+     */
+    @Test
+    public void testCaseInAggregationFunction() throws Exception {
+        String sqlQuery = String.format("SELECT SUM(100 - CASE WHEN value IS NULL THEN 0 ELSE 100 END) FROM '%s'", TEST_METRIC_NAME);
```
Comment (vtols) [inline src/test/java/com/axibase/tsd/api/method/sql/operator/SqlCaseOperatorTest.java]: The line is too long, try to split it in two or more, also concerns other long lines

**Comment by GregoryKutuzov** [comment]
it seems that we already have case tests class. Add new tests to it.

**Pull request merged**